process/headerCheck/headerVersioningHandler.go

```diff
@@ -88,20 +98,48 @@ func (hvh *headerVersioningHandler) GetVersion(epoch uint32) string {
 
 // GetVersion returns the version by providing the epoch
 func (hvh *headerVersioningHandler) getMatchingVersion(epoch uint32) string {
-	//TODO add a small cache here?
+	storedVersion, ok := hvh.getFromCache(epoch)
+	if ok {
+		return storedVersion
+	}
 
 	for _, ver := range hvh.versions {
 		if ver.StartEpoch <= epoch && epoch < ver.EndEpoch {
+			hvh.setInCache(epoch, ver.Version)
+
 			return ver.Version
 		}
 	}
 
 	log.Debug("headerVersioningHandler.GetVersion version not found",
 		"epoch", epoch, "default version", hvh.defaultVersion)
 
+	hvh.setInCache(epoch, hvh.defaultVersion)
+
 	return hvh.defaultVersion
 }
 
+func (hvh *headerVersioningHandler) getFromCache(epoch uint32) (string, bool) {
+	key := make([]byte, keySize)
+	binary.BigEndian.PutUint32(key, epoch)
```

Comment:
it would be much optimal to use epoch % 256 ?

Reply:
Since the byte slice conversion is quite cheap and we already have a small cache, I wouldn't venture to make this modulo operation since it has the same output for the epochs separated by multiple of 256, leading to wrong value outputs. I want to keep this as straight forward as it can get.